src/persist/src/location.rs

```diff
@@ -70,7 +70,8 @@ pub struct Determinate {
 
 impl std::fmt::Display for Determinate {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "determinate: {}", self.inner)
+        write!(f, "determinate: ")?;
+        fmt::Display::fmt(&self.inner, f)
```

Comment:
Convention is to use the method form here:

```suggestion
        self.inner.fmt(f)
```

Reply:
done